languages/zh-tw/constants.genie

```diff
@@ -0,0 +1,134 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of ThingTalk
+//
+// Copyright 2017-2018 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Giovanni Campagna <gcampagn@cs.stanford.edu>
+//
+// See COPYING for details
+
+{
+const assert = require('assert');
+
+const ThingTalk = require('thingtalk');
+const Ast = ThingTalk.Ast;
+const Type = ThingTalk.Type;
+const { clean } = require('../../lib/utils');
+
+// import the combinator library
+const C = require('../../lib/sentence-generator/ast_manip');
+}
+
+constant_Number = {
+    // `turking` is a flags that controls a group of grammar rules
+    // syntax is ?flag to enable the rule (or group of rules) only if the flag is set, !flag to disable the
+    // rule if the flag is set
+
+    !turking {
+    '0'    => new Ast.Value.Number(0);
+    '零' => new Ast.Value.Number(0);
+    '1'    => new Ast.Value.Number(1);
+    '一'  => new Ast.Value.Number(1);
+    }
+}
+
+constant_date_point = {
+    '現在'             => new Ast.Value.Date(null, '+', null);
+    '今天'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'day'), '+', null);
+    '昨天'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'day'), '-', Ast.Value.Measure(1, 'day'));
+    '這星期'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
+    '這週'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
+    '這禮拜'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
```

Comment:
According to <https://corenlp.run>, these should be two words.
For speed, all NL fragments in the templates are tokenized/segmented ahead of time, hence you'll need to split them with a space.

Reply:
I see. 
I was thinking about how it does Chinese segmentation lol
So the template we are going to do for device should be segment also right?

Reply:
No, primitive templates undergo segmentation by calling CoreNLP and should be written naturally.
Indeed, if you look at the `dataset.tt` that comes out of calling `genie download-dataset`, you'll see each example has `#_[utterances]` and `#_[preprocessed]` - developers provide the former in Thingpedia, and the latter is generated automatically by tokenizing `utterances`.

Reply:
I see. Thank you!